Tools/autotest/arducopter.py

```diff
@@ -2735,6 +2735,29 @@ def fly_rtl_speed(self):
         self.monitor_groundspeed(wpnav_speed_ms, tolerance=0.6, timeout=5)
         self.do_RTL()
 
+    def fly_rtl_alt_type_ascend(self, rtl_alt_m, rtl_alt_type=0):
+        self.context_push()
+        takeoff_alt = 15                # in m
+        self.set_parameters({
+            'RTL_ALT_TYPE': rtl_alt_type,
+            'RTL_ALT': rtl_alt_m * 100,   # m -> cm
+        })
+        self.reboot_sitl()
+        self.change_mode('GUIDED')
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+        self.takeoffAndMoveAway(takeoff_alt, 180)
+        self.set_rc_default()
+        self.change_mode('RTL')
+        self.wait_altitude(rtl_alt_m - 1, rtl_alt_m + 1, relative = not (rtl_alt_type==2))
+        self.wait_rtl_complete()
+        self.context_pop()
+
+    def fly_rtl_alt_type(self):
+        """Test RTL Alt Type parameter"""
+        self.fly_rtl_alt_type_ascend(25, 0)     # test ascend with above-home alt type
+        self.fly_rtl_alt_type_ascend(609, 2)    # test ascend with above-sea alt type
```

Comment:
We try to make these tests work regardless of where you start them - or make them start in a specific location.

In this case you should be able to replace "609" with a function call to get absolute initial alt and then add your relative margin.

Reply:
Done! I have changed it to get absolute inital alt and use it to calculate rtl alt (in abs frame).